content/usgs-packages/Application_Part3.Rmd

```diff
@@ -0,0 +1,182 @@
+
+# download precip data
+precip_stencil <- webgeom(paste0('HUC8::', paste(huc8s, collapse=",")))
+precip_knife <- webprocess() # accept defaults for weighted average
+all_webdata <- query("webdata")
+precip_fabric <- webdata(all_webdata["United States Stage IV Quantitative Precipitation Archive"])
+variables(precip_fabric) <- query(precip_fabric, 'variables')
+times(precip_fabric) <- c(startDate, endDate)
+```
+
+Executing the geojob:
+
+```{r execute-job-off, addToggle="ON", eval=FALSE}
+precip_geojob <- geoknife(precip_stencil, precip_fabric, precip_knife)
+wait(precip_geojob, sleep.time = 10) # add `wait` when running scripts
+precip_data <- result(precip_geojob)
+```
+
+```{r read-intermediate, echo=FALSE}
+geoknife_list <- readRDS("precip_geoknife.RDS")
+precip_geojob <- geoknife_list$geojob
+precip_data <- geoknife_list$data
+```
+
+## Prepare data for plotting
+
+First re-organize the precipitation into a long format (one column for value, one for HUC) rather than a separate column of precip for each HUC. See `gather` from the package `tidyr` for ideas on how to do it.
+
+```{r reorganize-precip-data, addToggle="ON"}
+library(tidyr)
+precip_data_long <- gather(precip_data, huc8, precip, -which(!names(precip_data) %in% huc8s))
+```
+
+##
+Now choose one site to create the plots. Seperate the data so each data frame contains only one site (see `?filter` from `dplyr` for hints). After we establish the plotting code, we will automate this for all sites. 
+
+```{r filter-data, addToggle="ON", message=FALSE, warning=FALSE}
+library(dplyr)
+
+nitrogen_site1 <- filter(nitrogen_data, site_no == sites[1])
+phosphorus_site1 <- filter(phosphorus_data, site_no == sites[1])
+
+huc_site1 <- filter(sb_sites_info, site_no == sites[1])$huc_cd # corresponding HUC8
+precip_site1 <- filter(precip_data_long, huc8 == huc_site1)
+```
+
+## Create time series plots
+
+Create a separate time series plot of precipitation, nitrogen, and phosphorus for each site using the separated data frames. See `?layout` to see how to include all three in one graphics device. 
+
+```{r time-series-plots, addToggle="ON"}
+layout(matrix(1:3, nrow=3))
+plot(precip_site1$DateTime, precip_site1$precip,
+     col="red", pch=20, xlab = "Time", ylab = "Precip accumulation, in")
+plot(nitrogen_site1$sample_dt, nitrogen_site1$result_va, 
+     col="green", pch=20, xlab = "Time", ylab = "Nitrogren concentration, mg/l")
+plot(phosphorus_site1$sample_dt, phosphorus_site1$result_va,
+     col="blue", pch=20, xlab = "Time", ylab = "Phosphorus concentration, mg/l")
+```
+
+## Automate the plots for all sites
+
+Now that we have established the code required to make all three timeseries plots for each site, automate this so it happens for each site (a.k.a. make a [for loop](/intro-curriculum/reproduce/#looping)). 
+
+```{r automate-time-series-plots, addToggle="ON"}
+for(i in sites){
+  huc_site_i <- filter(sb_sites_info, site_no == i)$huc_cd # corresponding HUC8
```

Comment:
:open_mouth: where has `dplyr::pull` been all my life?! I honestly did not know that was a function.